core/consensus/sumeragi.cpp

```diff
@@ -170,18 +171,50 @@ namespace sumeragi {
         std::string     myPublicKey;
         std::deque<std::unique_ptr<peer::Node>> validatingPeers;
 
+        Context()
+        {
+            update();
+        }
+
         Context(std::vector<std::unique_ptr<peer::Node>>&& peers)
         {
             for (auto&& p : peers) {
                 validatingPeers.push_back(std::move(p));
             }
         }
+
+        void update()
+        {
+            auto peers = config::PeerServiceConfig::getInstance().getPeerList();
+            for (auto&& p : peers ) {
+                validatingPeers.push_back( std::move(p) );
+            }
+
+            this->numValidatingPeers = this->validatingPeers.size();
+            // maxFaulty = Default to approx. 1/3 of the network.
+            this->maxFaulty = config::IrohaConfigManager::getInstance()
+                    .getMaxFaultyPeers(this->numValidatingPeers / 3);
```

Comment:
Should be: ```(this->numValidatingPeers-1)/3```. Also, there should be a check to set this to "0" if the result is less than 0.